### PR TITLE
Add support for remapping group-add

### PR DESCRIPTION
**Pull request by mrunalp**
@rhatdan @rhvgoyal PTAL.

A couple more enhancements that could be added are checking for group-add within initial range and coalescing group-add mappings if they are contiguous.

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by mrunalp** [comment]
I have added validation for 5 gid mappings limit.

**Comment by runcom** [comment]
Should this go upsteam?

**Comment by mrunalp** [comment]
@runcom No, longer term solution here is support for passing uid/gid mappings. This is just for OpenShift online.

**Comment by runcom** [comment]
:confused: I sort of see... Is this patch needed in other branches as well?

File: daemon/config_unix.go [hunk 1]
```diff
@@ -90,6 +91,7 @@ func (config *Config) InstallFlags(cmd *flag.FlagSet, usageFn func(string) strin
 	cmd.BoolVar(&config.EnableSecrets, []string{"-enable-secrets"}, true, usageFn("Enable Secrets"))
 	cmd.StringVar(&config.CgroupParent, []string{"-cgroup-parent"}, "", usageFn("Set parent cgroup for all containers"))
 	cmd.StringVar(&config.RemappedRoot, []string{"-userns-remap"}, "", usageFn("User/Group setting for user namespaces"))
+	cmd.BoolVar(&config.RemapGroupAdds, []string{"-userns-remap-group-adds"}, false, usageFn("Add user namespace gid mapping for group-adds"))
```
Comment (rhatdan) [inline daemon/config_unix.go]: Can you make this a hidden feature since we don't want our customers using it.

Comment (mrunalp) [inline daemon/config_unix.go]: @rhatdan Sure, any suggestion on approach?

Comment (runcom) [inline daemon/config_unix.go]: should be prefixing the flag with `#` to make it hidden like https://github.com/projectatomic/docker/blob/docker-1.12.6/daemon/config_unix.go#L89

Comment (mrunalp) [inline daemon/config_unix.go]: @runcom cool, I'll do that.

Comment (mrunalp) [inline daemon/config_unix.go]: Maybe we can add a comment that it isn't supported?

Comment (runcom) [inline daemon/config_unix.go]: prefixing with `#` should directly hide the flag in the --help. But sure, you can also comment that's disabled.

Comment (mrunalp) [inline daemon/config_unix.go]: Alright pushed an update and marked it as experimental in help text.

**Comment by rhatdan** [comment]
:+1:

**Comment by rhatdan** [comment]
What is the current state of this patch?  Should it be merged?

**Comment by rhatdan** [comment]
Still wondering if we should merge this?

**Comment by mrunalp** [comment]
We can close this for now.

**Pull request closed without merging**